src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -381,6 +405,30 @@ export default class AmalgamatingBusinesses extends Mixins(AmalgamationMixin, Co
     this.isAddingAmalgamatingForeignBusiness = false
   }
 
+  /**
+   * Check if business is already in table and display snackbar text.
+   * @param business The business being added.
+   * @param extraPro Whether we are adding an extra provincial company or not (A companies).
+   */
+  checkForDuplicateInTable (business: any, extraPro: boolean): boolean {
+    // Check if duplicate exists for a company that is not an extra-provincial.
+    const duplicateNonExtraPro = !extraPro &&
+      this.getAmalgamatingBusinesses.find(
+        (b: any) => (b.identifier && b.identifier === business.businessInfo.identifier))
+
+    // Check if duplicate exists for a company that is an extra-provincial (A company).
+    const duplicateExtraPro = extraPro &&
+      this.getAmalgamatingBusinesses.find(
+        (b: any) => (b.corpNumber && b.corpNumber === business.corpNumber))
+
+    if (duplicateNonExtraPro || duplicateExtraPro) {
+      this.snackbarText = 'Business is already in table.'
+      this.snackbar = true
```

Comment:
Since enabling the snackbar is otherwise done in `saveAmalgamatingBusiness()`, I'd move this there. (Keep like code together.)